src/scopes/utils.ts

```diff
@@ -109,3 +109,33 @@ export function filterAssociation(baseQuery, searchTerms, exclude, callback, com
 }
 
 export const idClause = (query: string[]) => query.filter((id: string) => !Number.isNaN(parseInt(id, DECIMAL_BASE))).join(',');
+
+/**
+ * Extracts the WHERE clause from a Sequelize model's findAll query and replaces the model name
+ * with an alias.
+ * @param model - The Sequelize model to query.
+ * @param alias - The alias to replace the model name with.
+ * @param scope - The WHERE options for the query.
+ * @returns The modified WHERE clause as a string.
+ */
+export const scopeToWhere = async (
+  model,
+  alias: string,
+  scope: WhereOptions,
+): Promise<string> => {
+  let sql = '';
+  // The db is not connected for this query as the limit is set to zero, it just returns.
+  await model.findAll({
+    where: scope,
+    limit: 0,
+    logging: (x) => { sql = x; },
```

Comment:
Sanity check - all the input ends up escaped right?

Reply:
I don't understand your question.

Reply:
nvm - I was trying to think if there was something that wouldn't be escaped, but I guess if you are just grabbing the SQL, everything should already have been take care of